Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1373 @@
+                    return ((LimitOrder)order).LimitPrice;
+                case OrderType.Market:
+                    // Order price must be positive for market order too;
+                    // refuses for price = 0
+                    return null;
+                case OrderType.StopLimit:
+                    return ((StopLimitOrder)order).LimitPrice;
+                case OrderType.StopMarket:
+                    return ((StopMarketOrder)order).StopPrice;
+            }
+
+            throw new NotSupportedException($"ZerodhaBrokerage.ConvertOrderType: Unsupported order type: {order.Type}");
+        }
+
+        /// <summary>
+        /// Return a relevant price for order depending on order type
+        /// Price must be positive
+        /// </summary>
+        /// <param name="order"></param>
+        /// <returns></returns>
+        private static decimal? GetOrderTriggerPrice(Order order)
+        {
+            switch (order.Type)
+            {
+                case OrderType.Limit:
+                    return ((LimitOrder)order).LimitPrice;
+                case OrderType.Market:
+                    // Order price must be positive for market order too;
+                    // refuses for price = 0
+                    return null;
+                case OrderType.StopLimit:
+                    return ((StopLimitOrder)order).StopPrice;
+                case OrderType.StopMarket:
+                    return ((StopMarketOrder)order).StopPrice;
+            }
+
+            throw new NotSupportedException($"ZerodhaBrokerage.ConvertOrderType: Unsupported order type: {order.Type}");
+        }
+
+        private static string ConvertOrderType(OrderType orderType)
+        {
+
+        //            public const string ORDER_TYPE_MARKET = "MARKET";
+        //public const string ORDER_TYPE_LIMIT = "LIMIT";
+        //public const string ORDER_TYPE_SLM = "SL-M";
+        //public const string ORDER_TYPE_SL = "SL";
+
+            switch (orderType)
+            {
+                case OrderType.Limit:
+                    return "LIMIT";
+                case OrderType.Market:
+                    return "MARKET";
+                case OrderType.StopMarket:
+                    return "SL-M";
+                case OrderType.StopLimit:
+                    return "SL";
+                default:
+                    throw new NotSupportedException($"ZerodhaBrokerage.ConvertOrderType: Unsupported order type: {orderType}");
+            }
+        }
+
+
+        /// <summary>
+        /// Updates the order with the same id
+        /// </summary>
+        /// <param name="order">The new order information</param>
+        /// <returns>True if the request was made for the order to be updated, false otherwise</returns>
+        public override bool UpdateOrder(Order order)
+        {
+            LockStream();
+
+            if(order.Status==OrderStatus.PartiallyFilled || order.Status == OrderStatus.Filled)
```

Comment:
Can't partially filled orders be updated?

Reply:
Yes it can be. We have updated that now